WordPress/src/main/java/org/wordpress/android/ui/posts/PrepublishingActionsFragment.kt

```diff
@@ -41,7 +41,7 @@ class PrepublishingActionsFragment : Fragment() {
     }
 
     private fun initViewModel() {
-        viewModel = ViewModelProviders.of(this, viewModelFactory)
+        viewModel = ViewModelProviders.of(this.requireParentFragment(), viewModelFactory)
```

Comment:
I am utilizing the `ParentFragment` here so that this `ViewModel` can share scope and the parent and the child fragments can communicate. Is this approach okay?

Reply:
yeah, this is a bit complicated :D. Ideally the two fragments would be independent of each other so they could be reused separately. However, I wouldn't complicate things unless necessary. 
One best practice which I think is important to adhere to is that the ParentViewModel shouldn't contain any logic related to the child fragment - it can provide interfaces so the two fragments can communicate, but it shouldn't be a replacement for a viewModel of the child fragment. I don't have enough context to say if we adhere to it. However, if we don't we could create a new child ViewModel and pass the parent ViewModel into it so they can communicate (in other words we'd ask the ViewModelProviders for two VM - one parent and one child). Wdyt?

Note: We might also consider communicating through an interface so the child doesn't have access to all the method of the parent, wdyt?

Reply:
Thanks for this comment @malinajirka 🙏 so I was just looking around the web to see the views others have of this and I am now seeing where a lot of people are using this shared viewmodel approach of communicating 

https://blog.mindorks.com/shared-viewmodel-in-android-shared-between-fragments
https://stackoverflow.com/questions/44272914/sharing-data-between-fragments-using-new-architecture-component-viewmodel

Just so you know the `ParentViewModel` doesn't contain any logic related to the child fragment. This shared approach just allows one `ViewModel` to be observed to trigger something on another `ViewModel`.

Let me know what you think about this as well. :)

Reply:
I'm sorry, but I'm not sure I follow. We are still talking about the same thing, right?

One parent (shared) view model which doesn't contain child logic, but provides a way of communication with the child.
One child (not-shared) view model which contains the child logic.

We use this approach for example on the PostListScreen - https://github.com/wordpress-mobile/WordPress-Android/blob/4585d19679b9f651d39dfd3372de544a093b221c/WordPress/src/main/java/org/wordpress/android/ui/posts/PostListFragment.kt#L88.

However, the current implementation confuses me a bit 
1. how come we need to call `vm.start` from the child fragment - haven't the parent already started the viewModel?
2. It seems the child fragment listens to UiState of the parent ViewModel - UiState is a logic and it seems weird that the parent is providing a UI state for the child. Wdyt?

Please let me know if I'm missing something and not answering your questions. We can perhaps chat about this or jump on a quick call and share the results of our discussion here. Wdyt?

Reply:
Yes, the approach I am taking seems identical to the `PostListFragment` but yes, we can jump on a call to discuss this. 🙏